aiidalab_widgets_base/structures.py

```diff
@@ -193,12 +215,16 @@ def _on_file_upload(self, change):  # pylint: disable=unused-argument
         self.file_path = os.path.join(tempfile.mkdtemp(), self.file_upload.filename)
         with open(self.file_path, 'w') as fobj:
             fobj.write(self.file_upload.data.decode("utf-8"))
-        structure_ase = get_ase_from_file(self.file_path)
-        self.on_structure_selection(structure_ase=structure_ase, name=self.file_upload.filename)
+        self.structure = get_ase_from_file(self.file_path)
+
+    @default('structure')
+    def _default_structure(self):
+        return None
```

Comment:
I'm not sure if this is necessary, but fine to leave.

Reply:
ok. I will keep.